boards: arm: Board Creation for cy8cproto_063_ble

sreeramIfx opened the pull request:
Adding support for Infineon cy8cproto_063_ble board

carlescufi commented:
@npal-cy

npal-cy commented:
Hi @sreeramIfx, you need to add dts\soc support for PSoC6 1M

DTS part:
dts\arm\infineon\psoc6\mpns\CYBLE_416045_02.dtsi (you can add all  PSoC6 1M MPNs)
dts\arm\infineon\psoc6\psoc6_01\*.*

SOC part:
\soc\arm\infineon_cat1\psoc6\Kconfig.defconfig.soc.psoc6_01
\soc\arm\infineon_cat1\psoc6\Kconfig.soc.psoc6_01

npal-cy commented on boards/arm/cy8cproto_063_ble/cy8cproto_063_ble.yaml:
```diff
@@ -0,0 +1,22 @@
+  - gpio
+  - uart
+  - i2c
+  - pwm output
```
let's enable supported feature once it added to mainline, so for:
```
supported:
  - gpio
  - uart
```

npal-cy commented on boards/arm/cy8cproto_063_ble/Kconfig.defconfig:
```diff
@@ -0,0 +1,12 @@
+# CY8CPROTO-062-4343W PSoC™ 6 Wi-Fi BT Prototyping Kit configuration
```
update board name "CY8CPROTO-063-BLE PSoC™ 6 BLE Prototyping Kit"

npal-cy commented on boards/arm/cy8cproto_063_ble/cy8cproto_063_ble_defconfig:
```diff
@@ -0,0 +1,38 @@
+CONFIG_CLOCK_CONTROL=y
+
+# Enable assert
+CONFIG_ASSERT=y
```
remove this CONFIG_ASSERT.

npal-cy commented on boards/arm/cy8cproto_063_ble/doc/index.rst:
```diff
@@ -0,0 +1,116 @@
+
+.. _PSoC 63 BLE MCU SoC Website: https://www.infineon.com/cms/en/product/evaluation-boards/cy8cproto-063-ble/
+
+.. _PSoC 63 BLE MCU Datasheet:
```
missed links to web resources...

sreeramIfx committed:
boards: arm: Board Creation for cy8cproto_063_ble

Add initial version of cy8cproto_063_ble board

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

galak reviewed (changes requested):
Can you reduce the duplication in the CY*.dtsi files.  Not 100% sure what the differences all are, but seems alot of them are just about mem/flash sizes.  In those cases, they dtsi should just update the reg property for those nodes.

galak commented on dts/arm/infineon/psoc6/mpns/CY8C6336BZI_BLD14.dtsi:
```diff
@@ -0,0 +1,21 @@
+/ {
+
+
+
```
can you cleanup the whitespace here (remove blank lines), and in other similar files.

sreeramIfx replied:
updated..

galak reviewed (changes requested):
Can you refactor the duplication on the dts/arm/infineon/psoc6/psoc6_01/psoc6*.dtsi files.  Looks like a fair amount of duplication under the pinctrl node.

npal-cy commented:
Hi @galak 

> Can you refactor the duplication on the dts/arm/infineon/psoc6/psoc6_01/psoc6*.dtsi files. Looks like a fair amount of duplication under the pinctrl node.

package dtsi (dts/arm/infineon/psoc6/psoc6_01/psoc6_01.xxxxx.dtsi) reflect gpio resources and pinctrl nodes which is suitable for this  packages. Example if device has psoc6_01.124-bga-sip.dtsi we declare pinctrl for uart_scb0...uart_scb7, if it  psoc6_01.43-smt.dtsi we uart_scb0...uart_scb6 (we do not add uart_scb7, uart_scb8 because in does not supported by this kind of package). Let me know if it makes sense.

ifyall commented:
@galak, do you have any more questions on the DTSI files here?

sreeramIfx committed:
soc: infineon_cat1: Support PSoC 1M devices

- Add the files required to support PSoC 1M devices

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

galak reviewed (changes requested):
Some of the shared nodes in in the psoc6_01.xxx*dtsi files for things like hsiom, gpio, etc should be in a common dtsi file.

galak reviewed (changes requested):
The uart/i2c/spi/... as a scb node and than in the board dts it modifies the compatible for i2c/spi/uart.

Look at ATMEL sercom or NXP flecomm to see examples:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/adafruit_feather_m0_lora/adafruit_feather_m0_lora.dts#L43

Here's a rough example of what the dtsi:

```
		scb4: scb@40650000 {
			compatible = "infineon,cat1-scb";
			reg = <0x40650000 0x10000>;
			#address-cells = <1>;
			#size-cells = <0>;
			interrupts = <45 6>;
			status = "disabled";
		};
		
		scb5: scb@40660000 {
			compatible = "infineon,cat1-scb";
			reg = <0x40660000 0x10000>;
			#address-cells = <1>;
			#size-cells = <0>;
			interrupts = <46 6>;
			status = "disabled";
		};	
```

sreeramIfx commented:
> The uart/i2c/spi/... as a scb node and than in the board dts it modifies the compatible for i2c/spi/uart.
> 
> Look at ATMEL sercom or NXP flecomm to see examples: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/adafruit_feather_m0_lora/adafruit_feather_m0_lora.dts#L43
> 
> Here's a rough example of what the dtsi:
> 
> ```
> 		scb4: scb@40650000 {
> 			compatible = "infineon,cat1-scb";
> 			reg = <0x40650000 0x10000>;
> 			#address-cells = <1>;
> 			#size-cells = <0>;
> 			interrupts = <45 6>;
> 			status = "disabled";
> 		};
> 		
> 		scb5: scb@40660000 {
> 			compatible = "infineon,cat1-scb";
> 			reg = <0x40660000 0x10000>;
> 			#address-cells = <1>;
> 			#size-cells = <0>;
> 			interrupts = <46 6>;
> 			status = "disabled";
> 		};	
> ```

@galak we will have to make these changes to the already merged [psoc6_02.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi) as well. Can we make these changes in a separate PR? We want to review the implementations of ATMEL/NXP and work on our implementation as a separate task.

sreeramIfx committed:
dts: infineon: Optimize node declarations

Move common declarations to the parent dtsi file

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx commented:
> Some of the shared nodes in in the psoc6_01.xxx*dtsi files for things like hsiom, gpio, etc should be in a common dtsi file.

moved the gpio, hsiom nodes to a common file and used /delete-node/ to configure per package..

galak commented:
> @galak we will have to make these changes to the already merged [psoc6_02.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi) as well. Can we make these changes in a separate PR? We want to review the implementations of ATMEL/NXP and work on our implementation as a separate task.

We should figure this out now.  The impact to psoc6_02.dtsi seems pretty small at this point since it only seems to be defining uart's and not all the other I2C, SPI, etc. nodes.

sreeramIfx committed:
dts: infineon: SCB declaration

Declare SCB nodes to be used as UART/SPI/I2C by the boards

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx committed:
dts: infineon: SCB declaration

Declare SCB nodes to be used as UART/SPI/I2C by the boards, Move
common declarations from psoc6_02 to the parent dtsi file

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx committed:
dts: infineon: Support for PSoC 1M devices

Add Device tree files for PSoC 1M devices

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx committed:
drivers: flash: Fix kconfig dependency

- Update KCONFIG for flash driver to depend on
DT_HAS_FIXED_PARTITIONS_ENABLED, since that is necessary for the
driver implementation

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

galak reviewed (approved):
LGTM.  Thanks.

We should follow up with updating the i2c, uart, etc bindings to include the `infineon,cat1-scb.yaml` to convey the relationship.

sreeramIfx commented on boards/arm/cy8cproto_063_ble/doc/index.rst:
```diff
@@ -0,0 +1,126 @@
+
+The default configurations can be found in the Kconfig
+
+:zephyr_file:`boards/arm/cy8cproto-063-ble/cy8cproto_063_defconfig`
```
updated @ https://github.com/zephyrproject-rtos/zephyr/pull/57814

sreeramIfx commented:
> 

Updated @ https://github.com/zephyrproject-rtos/zephyr/pull/57814
